src/persist-client/tests/machine/compaction_single

```diff
@@ -0,0 +1,147 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+## Compaction with a single nonempty input is special cased to
+## return the nonempty batch directly with a new description
+
+# always compact if our only nonempty batch has a since of [0], in order to trim truncatable data
+write-batch output=b lower=1 upper=2
+zero 1 1
+----
+parts=1 len=1
+
+compact output=b_compacted inputs=(b) lower=1 upper=2 since=99
+----
+parts=1 len=1
+
+fetch-batch input=b_compacted
+----
+<part 0>
+zero 99 1
+<run 0>
+part 0
+
+# attempting to compact our just-compacted batch should hit our special case,
+# as it is the single nonempty batch with since in advance of [0]
+compact output=b_compacted_again inputs=(b_compacted) lower=1 upper=2 since=999
+----
+parts=1 len=1
+
+# (the since has not advanced, confirming we're reading the same contents referred by b_compacted)
```

Comment:
nit: s/since/timestamp/ here and elsewhere

Reply:
eeek good catch